### DISCUSSION TITLE
Is there a batch API to trigger a job?

### DISCUSSION THREAD

**Discussion by andrewtsw** [quartznet/quartznet#1809] [Q&A] [answered]
Hello.
I want to trigger the same job about 1000 times with different values in `JobDataMap`
I' use `scheduler.TriggerJob` for now. It works but it triggers each job in a separate transaction.
Is there a batch API to trigger job?

Like 
```
Task TriggerJob(
            JobKey jobKey, 
            IEnumerable<JobDataMap> data,
            CancellationToken cancellationToken = default);
```

**Comment by lahma** [comment, marked as answer]
If you look how `TriggerJob` is implemented, it basically just creates a simple trigger against the given job.

https://github.com/quartznet/quartznet/blob/cc953029cd1406cb2a9f7a4afc16d9ea1ff9e35a/src/Quartz/Core/QuartzScheduler.cs#L939-L972

With `ScheduleJob` overload you should be able to give the 1000 triggers with different job data maps.

https://github.com/quartznet/quartznet/blob/cc953029cd1406cb2a9f7a4afc16d9ea1ff9e35a/src/Quartz/IScheduler.cs#L347-L351
